File: string/truncate-sentence/truncate-sentence.go
package truncatesentence

import "strings"

func truncateSentenceEasy(s string, k int) string {
	words := strings.Split(s, " ")
	words = words[:k]

	return strings.Join(words, " ")
}

func truncateSentence(s string, k int) string {
	for i, c := range s {
		if c == ' ' {
			k--
			if k == 0 {
				return s[:i]
			}
		}
	}

	return s
}

File: string/truncate-sentence/truncate-sentence.ts
function truncateSentenceEasy(s: string, k: number): string {
    return s.split(" ").slice(0, k).join(" ");
};


function truncateSentence(s: string, k: number): string {
    for (let i = 0; i < s.length; i++) {
        if (s[i] === ' ' && --k === 0) {
            return s.slice(0, i);
        }
    }

    return s;
};
